### DISCUSSION TITLE
EZ1-M accuracy

### DISCUSSION THREAD

**Discussion by Marcwa19197** [SonnenladenGmbH/APsystems-EZ1-API-HomeAssistant#10] [General]
Hi,

first of all: Thank you for the very nice integration!

Im using the inverter since a few days now. Does someone know how accurate the production data is?
On the outlet where the inverter is connected to i also use a Shelly Plus 1PM which shows different data. So what data tells the truth?

Which data is more accurate? Shelly or the inverter itself?

Example:
Shelly shows 0,006399 kWh of total generated energy.
Inverter shows 0,02218 kWh of total generated energy. (Reading sensor.apsystems_nameofdevice_lifetime_production)

Thanks!

**Comment by mawoka-myblock** [comment]
I really think that this is negligible. If you'd have a calibrated trustworthy device to measure against, I'd understand it, but I wouldn't consider the Shelly plug a trustworthy measuring device, especially not with low currents.

**Reply by Marcwa19197** [reply]
Yes - i also read that shellys are not accurate. But is the data from the inverter more accurate? Im just asking to add the correct sensor data to the Energy-Dashboard in HomeAssistant... 

or other question... if inverter data is "more accurate": Does it make sense to have the shelly in between? Maybe for an "Kill Switch" for remotely resetting the inverter?

**Reply by mawoka-myblock** [reply]
I can't say anything about the accuracy of the inverter, but I personally would trust the inverter more.

**Reply by Marcwa19197** [reply]
Okay - thanks! Will the "daily data" resets every 24h to 0? So which one is the best to use in energy-dashboard?

**Reply by mawoka-myblock** [reply]
Just use the daily production. That works. I don't know when it resets, but that also doesn't really matter, since it's, when the inverter is offline.
